tools/python/boutiques/validator.py

```diff
@@ -58,6 +62,82 @@ def inById(i):
             return descriptor["inputs"][inputGet("id").index(i)]
         return {}
 
+    def conditionalExpFormat(s):
+        # ex: "(opt1>2)" becomes " ( opt1 > 2 ) "
+        # "(opt1<=10.1)" becomes " ( opt1 <= 10.1 ) "
+        cleanedExpression = ""
+        idx = 0
+        while idx < len(s):
+            c = s[idx]
+            if c in ['=', '!', '<', '>']:
+                cleanedExpression += " {0}{1}".format(
+                    c, "=" if s[idx+1] == "=" else " ")
+                idx += 1
+            elif c in ['(', ')']:
+                cleanedExpression += " {0} ".format(c)
+            else:
+                cleanedExpression += c
+            idx += 1
+        return cleanedExpression
+
+    def isValidConditionalExp(exp):
+        # Return the type of a conditional expression's substring
+        def getSubstringType(s):
+            s = s.strip()
+            if s in schema_types:
+                return s
+            elif re.search(r'^[0-9]+\.?[0-9]*$', s):
+                return "Number"
```

Comment:
An additional type check we could do since the schema has a boolean for `integer = true/false` is check in the comparison whether or not the input that is of type number is an int or a float, and then also allowing us to be a bit more specific here and additionally separate floats from ints.

Reply:
I think we can allow comparison between floats and integers. If an ```NUM1``` input with ```integer = false``` evaluates to 9.5, the conditional path template with the key ```NUM1 <= 10``` should still be selected. What do you think?

Reply:
Yeah - you're right. How about in the case where `integer = true`? I think you're right, but we can just open an issue for this later if we think it is worth bringing back up :) Thanks!

Reply:
Okay, I still implemented the type check for Integer if we need it (ea90f62e426ae449e4ccb82de8c5570c7cbd17b0). I reverted it on the last commit because I thought it was too strict.